keps/sig-cli/2299-kustomize-plugin-composition/README.md

```diff
@@ -41,99 +44,121 @@
+- Change the flags and parameters used for invoking plugins in any way.
 
 ## Proposal
 
-Introduce `Composition` as a new API `kind` recognized by `kustomize build`. A `Composition` must be defined in a `composition.yaml` file, and inclusion of both `composition.yaml` and `kustomization.yaml` in the target directory of a build will be considered an error.
+Introduce `Composition` as a new API `kind` recognized by `kustomize build`. A `Composition` must be
+defined in a `composition.yaml` file. Inclusion of both `composition.yaml` and `kustomization.yaml`
+in the target directory of a build will be considered an error.
 
-The Composition API enables users to:
+The `Composition` API enables users to:
 
-- Define a list of Kubernetes-style configuration objects called **modules**.
-A module is a client-side resource that expresses the desired state implemented by
-a Kustomize plugin.
-- Import modules from another Composition and add them to the list.
-- Override an imported module's fields with new values.
-- Reorder the list of modules prior to execution.
+- Define a list of transformer configuration that intermingles built-in and plugin transformers.
+- Import transformers from another `Composition` and add them to the list.
+- Override an imported transformer's fields with new values.
+- Reorder the list of transformers prior to execution.
 
-Once a user has written a Composition, all they need to do is run `kustomize build`
-to turn it into a list of Kubernetes resources that they can commit to git.
+Once a user has written a `Composition`, they can run `kustomize build` to turn it into a list of
+Kubernetes resources that they can commit to git, the same way they would with `Kustomization`.
 
 To provide this experience, Kustomize will need to do the following during builds:
 
-- Consolidate the Composition and its imports into a finalized list of modules.
-- Automatically fetch and execute the plugin that implements each module.
-- Pass the output of each plugin invocation as the input to the next plugin.
+- Consolidate the `Composition` and its imports into a finalized list of transformers (which may be
+  plugins).
+- Execute each transformer in sequence (no changes to plugin execution are proposed by this KEP).
+- Pass the output of each transformer as the input to the next.
 
 ### User Stories (Optional)
 
 #### Story 1
 
-As a user, I can define my configuration as a sequence of declarative abstractions
-implemented by plugins that are automatically discovered and installed.
+As a user, I can define my configuration as a sequence of declarative abstractions implemented by
+plugins.
 
 ```yaml
 # app/composition.yaml
 kind: Composition
 
-modules:
+transformers:
 # generate resources for a Java application
 - apiVersion: example.com/v1
   kind: JavaApplication
-  provider: {container: {image: example/module_providers/java:v1.0.0}}
+  provider: {container: {image: docker.example.co/kustomize_transformers/java:v1.0.0}}
```

Comment:
Apologies if this is addressed elsewhere (I could not find it), but is this provider configuration intended to match the provider configuration proposed by catalog?

Reply:
This is just https://github.com/kubernetes-sigs/kustomize/blob/master/kyaml/fn/runtime/runtimeutil/functiontypes.go#L128, which currently surfaces as an annotation containing JSON data. This KEP is proposing that that information be moved from the annotation to this reserved field, at least within Composition. It is not proposing any particular changes to the data within.